### DISCUSSION TITLE
Design a token such that it's body can be easily queried as it's own token

### DISCUSSION THREAD

**Discussion by Guiltybyte** [tree-sitter/tree-sitter#1967] [Q&A] [answered]
My `grammer.js` defines a token "template" as follows:  
```
  template: $ => token(seq(
    choice('<:', '<:='),
    /.*/,
    ':>'
  )),

```  
When I run `treesitter parse foo.txt` any string beginning with `<:` or `<:=` and ending with `:>` are marked as being this template
token.
However the markers:  `<:`, `<:=`,  `:>`  are considered part of this `template` token which is correct, however I would like to structure this token so the "body" i.e. anything inside of the "markers" is it's own token. Mainly I want to do this to enable people to write queries as follows:

```
((template
   (body) @bar))
```
Which would use the `@bar` capture on only the token which encompasses everything within the markers, but excluding them.

How should I go about doing this?

**Comment by Guiltybyte** [comment]
Looking at the docs i thought that `field` function might be what I need, thus I changed my def to:
```
  template: $ => token(seq(
    choice('<:', '<:='),
    field('body', /.*/),
    ':>'
  )),

```

However when I parse the example file, the template node doesn't have any children, lest one named 'body' as i would have expected.

**Reply by ahelwer** [reply, marked as answer]
Break `/.*/` into its own rule.

**Reply by Guiltybyte** [reply]
Ok so I have done the following:
```
  body: $ => /.*/,

  template: $ => seq(
    choice('<:', '<:='),
    $.body,
    ':>'
  ),
```

But now `tree-sitter generate` hangs forever

**Reply by ahelwer** [reply]
`/.*/` is kind of a bad regex, it will just read until the end of the file. You probably want `/.*:>/` (idk whether you need to escape `:` and `>`) although then `:>` will be included in `body`.

**Reply by Guiltybyte** [reply]
hmm if I just leave it the way it is (i.e. as listed in my first post), would there be a way of using a query to extract just the so-called body?

**Reply by ahelwer** [reply]
I don't think so, no.
